### ISSUE TITLE
User Trigger `source` is not correctly localized

### ISSUE THREAD

**Issue by trim21** [quisquous/cactbot#4910]
## Description

I'm writing user trigger on `zoneId: ZoneId.DragonsongsRepriseUltimate` (Ultimate of patch 6.1)

I'm using a **CN** player.

and copy a `netRegex` from built-in trigger

```ts
    {
      id: 'DSR Wrath Thunderstruck',
      // This is effectId B11, but the timing is somewhat inconsistent based on statuses rolling out.
      // Use the Chain Lightning ability instead.
      type: 'Ability',
      netRegex: NetRegexes.ability({ id: '6B8F', source: 'Darkscale' }),
      // Call this after, which is ~2.3s after this ability.
...
    }
```

to my trigger

```ts
    {
      id: 'DSR P5 一运 雷点名',
      type: 'Ability',
      netRegex: NetRegexes.ability({ id: '6B8F', source: 'Darkscale' }),
...
    }
```

But this never get trigged.

Then I have to replace the source manually to `'暗鳞黑龙'` to make it work.

https://github.com/quisquous/cactbot/blob/d7084ff27230cddb44f8b9e7d73568cbfe4d46f3/ui/raidboss/data/06-ew/ultimate/dragonsongs_reprise_ultimate.ts#L3110-L3112

cactbot version：[2ef248df](https://github.com/quisquous/cactbot/commit/2ef248df)

you can find my log here: https://github.com/trim21/actions-cron/releases/tag/archive named [cactbot-issue-4910-log.log](https://github.com/trim21/actions-cron/releases/download/archive/cactbot-issue-4910-log.log)

**Comment by quisquous** [comment]
Looking at popup-text.ts, it looks like for timelines itself, everything is pushed into a `replacements` variable and so if the main trigger file has a timeline replacement then the user trigger file will use that replacement for timelines too.  However, for triggers the calls to `translateRegex` only use `set.timelineReplace` (the replacements for the trigger set/file itself) and so the user file would have to have the replacement itself right now (which seems wrong).

I think this could be fixed by moving these lines up higher (~line 676 ish or so):
```ts
      if (set.timelineReplace)
        replacements.push(...set.timelineReplace);
```

... and then using `replacements` instead of `set.timelineReplace` in the `translateRegex` calls.

**Comment by trim21** [comment]
@MaikoTan has a idea to replace net regex function call to its parameters considering we have type on all triggers already, could also simplifying localization

**Comment by quisquous** [comment]
Sorry that I don't understand, but I don't think this will fix this issue or help in localization?  I feel like the issue is what I described above.  Do you disagree?

**Comment by trim21** [comment]
No, I didn't disagree with you ( I'm saying this because I don't know how our localizetion works on regex in details now. )

I just thought that with this idea is good and may affect our i18n and mention here, with this new approach we can translate paramater first then building the regex instead other way around. It may be off-topic for this issue.

**Comment by quisquous** [comment]
Ok, makes sense.  Just wanted to make sure I understood the context.  I agree that it's probably cleaner to translate the parameters directly than to translate the entire regular expression.  It hasn't been an issue so far, and that's probably because the things that aren't translated (e.g. ids) just don't end up being anything that other replacement text would hit.  But, maybe there's edge cases, and it'd be safer to do it on parameters directly.

**Comment by trim21** [comment]
I AFK for a while and my memory is a bit fuzzy, is this already fixed by refacting net regex building?

**Comment by quisquous** [comment]
Reading the code, I think all `netRegex` (regardless of if it's parameters or helper function) should be translated.

**Comment by trim21** [comment]
they already are... I think?
